packages/multichain-account-service/src/MultichainAccountWallet.ts

```diff
@@ -355,4 +365,89 @@ export class MultichainAccountWallet<
       this.#isAlignmentInProgress = false;
     }
   }
+
+  /**
+   * Discover and create accounts for all providers.
+   *
+   * NOTE: This method should only be called on a newly created wallet.
+   *
```

Comment:
I think this is no longer true? We can run discovery on any wallets now since we're starting from its highest known group index?
```suggestion
```